src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const stateClones = [];
+
+  for (const action of actions) {
+    const newState = { ...currentState };
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        break;
+      case 'removeProperties':
+        action.keysToRemove.forEach(proper => {
+          delete newState[proper];
+        });
+        break;
+      case 'clear':
+        Object.keys(newState).forEach(proper => {
+          delete newState[proper];
+        });
```

Comment:
if `currentState` it `let` constant - you can do it easy
```suggestion
        currentState = {};
```

Reply:
Bc you do this
Remove it
![image](https://github.com/mate-academy/js_stateful-clones/assets/91411692/a6f5c03b-b005-4587-98b3-4de4d843081f)

Reply:
![image](https://github.com/mate-academy/js_stateful-clones/assets/122682570/99e5229d-89c4-49a1-a7e4-ff671647b265)
ось результат.

Reply:
Bc if you push `newState` - you should make `newState = {}`